### PR TITLE
Add clusters endpoint

### PR DESCRIPTION
**Pull request by palash25**
### Required for all PRs:

- [x] README.md updated (if needed)
- [x] Passes `gofmt` and `golint`
- [x] Fetch cluster status
- [x] Delete cluster
- [x] Create a cluster (both interactive and non-interactive modes) 

**Interactive mode screencast link:** https://asciinema.org/a/211437
- Interactive mode uses a dependency https://github.com/AlecAivazis/survey/
- Most questions require user input but a few like addons let the user select the value from a list of options and in case of plugins they can select multiple options
- Some questions also have help text
- I cannot find a workaround for how to take the input for the env field and right the user will have to type out the exact value of env along with the braces and everyting. (We can fix this later or even open a goo-first-issue for this)

**Non-interactive mode screencast gif:** 
non-interactive

### CODE REVIEW

**Comment by siddharthlatest** [comment]
@palash25 Is this ready to be merged?

**Comment by palash25** [comment]
Yup I was just waiting for utsav's review on this. You can see it working (interactive mode) on the screencast

**Comment by utsavoza** [comment]
@palash25 Can you also add the docs, I am not really sure about the usage of the command?

**Comment by palash25** [comment]
@utsavoza docs added

File: appbase/cluster/cluster_options.go [hunk 1]
```diff
@@ -129,8 +130,11 @@ var esOptions = []*survey.Question{
 		Validate: survey.Required,
 	},
 	{
-		Name:     "volume_size",
-		Prompt:   &survey.Input{Message: "Enter the volume size"},
+		Name: "volume_size",
+		Prompt: &survey.Select{
+			Message: "Enter the volume size",
+			Options: []string{"30", "40", "80", "160"},
```
Comment (palash25) [inline appbase/cluster/cluster_options.go]: These are the disk sizes already divide by the number of nodes.

**Comment by utsavoza** [comment]
@palash25, I was testing out the command and so far faced the following issues:
- Outdated pricing plans (check the docs once).
-  An issue during interactive mode, when navigating using arrow keys prints the same question repeatedly.
- Message regarding why cluster creation failed.

Since the pricing plans provided are outdated, the cluster creation fails currently. The command usage is printed before the command is terminated. It would be helpful if the user is aware of the reasons why the cluster creation was failed. The reason is that the options provided here are quite extensive and some are dependent on prior selected options (for example, selection the locations that are available based on provider, so if user selects a location provided by azure for gke provider, he should know why the deployment failed at the end of the command). One way to solve the problem would be to allow user to select the provider first, and based on that selection, we display the available regions for the provider, similar to what we do it from the dashboard. As far as I can tell, the backend returns the reason for why a cluster was failing, and we should let the user know about it.

Elaborating on the second point, during the interactive mode, in the question `How many add-ons would you like to add to your cluster deployment?`, when I used the arrow keys to move the cursor up/down to navigate the options, the question was getting printed to the console repeatedly. Similar issue occurred when I was navigating `Choose an addon from the following list:` prompt.

@raazcrzy Please do let us know if we are missing out on any other updates on the backend that might be affecting the command.

**Comment by palash25** [comment]
> when I used the arrow keys to move the cursor up/down to navigate the options, the question was getting printed to the console repeatedly.

@utsavoza  yeah that is a pain point, there is an open issue for this at the repo of the dependency that I am using for interactive mode https://github.com/AlecAivazis/survey/issues/101

however it doesn't happen if the terminal's width is a long as the question, so if the user has a maximized terminal or the terminal open in an IDE like vscode they might never experience it. I will try to see if I can shorten some of the questions to minimize the chances of this problem occurring but I guess we are stuck with this kind of behavior till the maintainer of the package fixes it and we update the dependency.

**Comment by palash25** [comment]
@utsavoza fixed the issue with provider based options, now options such as location are asked based on the provider selected by the user, the pricing_plan filed values were changed too so updated them as well, also an error message is displayed in case a cluster creation fails.

Its ready for review again :smile:

**Comment by utsavoza** [comment]
@palash25 The command works fine now. There is still one issue that I am facing currently and I am not sure if it's an issue in the command or the backend. For `gke` clusters, I am not able to create a cluster with VM size other than `custom-2-4096`. I keep on seeing this error message: `Invalid machine size provided for the specified pricing plan in request body`. @raazcrzy Is there a mapping between pricing plan and the appropriate VM size for `gke` clusters?

P.S: @palash25 Could you also update the docs so that they reflect the changes you have made in the previous commit?

**Comment by palash25** [comment]
@utsavoza I remember @raazcrzy told me on slack that `custom-2-4096` only works for Hobby and Sandbox plans so maybe you used one of the other machines with the Hobby and Sandbox plans. 

I am not sure about this either, because I didn't find anything in the API docs regarding machines and pricing plan mappings, if by any chance it is documented and I have missed is please let me know @raazcrzy 

> Could you also update the docs so that they reflect the changes you have made in the previous commit?

sure I will push another commit for docs and then ping you.

**Comment by utsavoza** [comment]
@palash25 @raazcrzy, I tried creating the cluster for all the pricing plans and VM sizes, but I wasn't able to do so for VM size other than `custom-2-4096`. Apparently, I was also able to create a `Production-I` cluster with `custom-2-4096` VM size, but am still unclear about why it is failing for other options.

**Comment by palash25** [comment]
@utsavoza found two mistakes I had left out two deprecated vm_sizes `g1-small` and `n1-standard-1` just removed those two. I am trying out the other combinations now.

**Comment by palash25** [comment]
@utsavoza plan to VM size mappings added, gke clusters with different plans are working now. 

Docs don't really need any updates since the interactive command is still the same.

**Comment by palash25** [comment]
docs updated @utsavoza can we merge this?

**Comment by utsavoza** [comment]
Changes looks good to me. 👍

**Pull request merged**